framework/include/postprocessors/NumFailedTimeSteps.h

```diff
@@ -0,0 +1,32 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "TimeStepper.h"
+#include "GeneralPostprocessor.h"
+
+/**
+ * This postprocessor outputs the amount of failed time steps.
+ */
+class NumFailedTimeSteps : public GeneralPostprocessor
+{
+public:
+  static InputParameters validParams();
+
+  NumFailedTimeSteps(const InputParameters & parameters);
+
+  void initialize() override {}
+  void execute() override {}
+  Real getValue() override;
+
+protected:
+  /// Retrieves the current TimeStepper used by the executioner.
+  TimeStepper * _timestepper;
```

Comment:
```suggestion
  const TimeStepper * const _timestepper;
```

Reply:
it might complain with the way it s initialized rn, let s see

Reply:
doesn't like this due to how its acquired -- compiler says that _app->getExecutioner()->getTimestepper is inherently not const, even if you declare it so. Think this will have to go.

Reply:
you should be able to add const-ness even without a cast
can you paste the error?

Reply:
```
/Users/peatna/moose/4moose/moose/framework/src/postprocessors/NumFailedTimeSteps.C:28:18: error: cannot assign to non-static data member '_timestepper' with const-qualified type 'const TimeStepper *const'
    _timestepper = dynamic_cast<Transient *>(_app.getExecutioner())->getTimeStepper();
    ~~~~~~~~~~~~ ^
/Users/peatna/moose/4moose/moose/framework/build/header_symlinks/NumFailedTimeSteps.h:31:29: note: non-static data member '_timestepper' declared const here
  const TimeStepper * const _timestepper;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
```

Reply:
It also wants you to explicitly initialize it if its const, so the _timestepper(NULL) would have to remain

Reply:
can you try
```
const TimeStepper *
```
pointer wont be const, but it ll point to a const object

Reply:
seems to have worked